### PR TITLE
Transient users: UX improvements

### LINKED ISSUE #24279: Transient users: UX
### Description

#23977 Introduced transient users for brokering preview feature.

This task is to improve UX for the transient users feature. Currently there are at least the following issues with UX:

- In the admin console:
   - Transient users from brokering should be distinguishable from regular persistent users, e.g. by using a badge or label 
   - Upon logging out transient user from user detail, the admin UI crashes since the user is deleted along the way and user endpoint returns 404

### Discussion

See discussion in https://github.com/keycloak/keycloak/pull/23977

### Motivation

#11334

### Details

_No response_

### PR DESCRIPTION
**Pull request by hmlnarik**
This PR delivers UI for transient users for brokering feature. For transient users:

- Label `Transient` is added next to the user name in user detail
- Logout from the session tab of a user detail removes the user. Thus the browser is redirected to user list after logout
- Deleting a transient user is done by user logout (which effectively removes the user)

TODO: Tests

Closes: #24279

### CODE REVIEW

File: js/apps/admin-ui/src/user/utils.ts [hunk 1]
```diff
@@ -25,3 +25,6 @@ export const fieldName = (attribute: UserProfileAttributeMetadata) =>
   isRootAttribute(attribute.name)
     ? attribute.name
     : `attributes.${attribute.name}`;
+
+export const isLightweightUser = (userId?: String) =>
```
Comment (jonkoops) [inline js/apps/admin-ui/src/user/utils.ts:29]: For primitive values in TypeScripts (e.g. `number`, `string`, etc), we use the lowercase variant.
```suggestion
export const isLightweightUser = (userId?: string) =>
```

File: js/apps/admin-ui/src/user/EditUser.tsx [hunk 2]
```diff
@@ -185,6 +194,24 @@ export default function EditUser() {
...
         className="kc-username-view-header"
         divider={false}
+        badges={
+          lightweightUser
+            ? [
+                {
+                  text: (
+                    <Tooltip content={t("transientUserTooltip")}>
+                      <Label
+                        data-testid={"user-details-label-transient-user"}
```
Comment (jonkoops) [inline js/apps/admin-ui/src/user/EditUser.tsx:204]: String literals can be passed directly.
```suggestion
                        data-testid="user-details-label-transient-user"
```

**Review by jonkoops — CHANGES_REQUESTED** [review]
Couple of small nits, otherwise LGTM. Looks like there are some failing test though.

**Comment by hmlnarik** [comment]
I believe the PR is now ready to go, so switching from draft to full.
@jonkoops Thank you for the review, could you please re-review?

**Review by jonkoops — APPROVED** [review]
LGTM

**Pull request merged**